shared/hooks/BackendAPI/useGet.js

```diff
@@ -5,37 +5,44 @@ import { useFetch } from './useFetch';
 
 const useGetHook = processDataFn =>
   function(path, { pollingInterval, onDataReceived, skip } = {}) {
-    const isHookMounted = React.useRef(true); // becomes 'false' after the hook is unmounted to avoid performing any async actions afterwards
     const lastAuthData = React.useRef(null);
     const lastResourceVersion = React.useRef(null);
     const [data, setData] = React.useState(null);
     const [loading, setLoading] = React.useState(true);
     const [error, setError] = React.useState(null);
     const { authData } = useMicrofrontendContext();
     const fetch = useFetch();
+    const abortController = React.useRef(new AbortController());
 
     const refetch = (isSilent, currentData) => async () => {
-      if (skip || !authData || !isHookMounted.current) return;
-      if (!isSilent) setLoading(true);
+      if (skip || !authData || abortController.current.signal.aborted) return;
+      if (!isSilent) setImmediate(_ => setLoading(true));
+
+      abortController.current = new AbortController();
 
       function processError(error) {
-        console.error(error);
-        if (!isSilent) setError(error);
+        if (!abortController.current.signal.aborted) {
+          console.error(error);
+          setError(error);
+        }
       }
 
       try {
-        const response = await fetch({ relativeUrl: path });
+        const response = await fetch({ relativeUrl: path, abortController });
         const payload = await response.json();
 
-        if (!isHookMounted.current) return;
+        if (abortController.current.signal.aborted) return;
         if (typeof onDataReceived === 'function') onDataReceived(payload.items);
-        if (error) setError(null); // bring back the data and clear the error once the connection started working again
-        processDataFn(payload, currentData, setData, lastResourceVersion);
+        if (error) setImmediate(_ => setError(null)); // bring back the data and clear the error once the connection started working again
+        setImmediate(_ =>
+          processDataFn(payload, currentData, setData, lastResourceVersion),
+        );
       } catch (e) {
-        processError(e);
+        setImmediate(_ => processError(e));
```

Comment:
```This feature is non-standard and is not on a standards track. Do not use it on production sites facing the Web: it will not work for every user. There may also be large incompatibilities between implementations and the behavior may change in the future.```

Reply:
```This method is not expected to become standard, and is only implemented by recent builds of Internet Explorer and Node.js 0.10+. It meets resistance both from Gecko (Firefox) and Webkit (Google/Apple).```

Reply:
Do we have to use this function?

Reply:
Well, I couldn't find any other solution...
You're right, most browsers [refuse to support it ](https://caniuse.com/setimmediate)(although it works well in Chrome and Firefox; somebody's lying here...)